examples/jsm/renderers/common/Renderer.js

```diff
@@ -805,17 +805,47 @@ class Renderer {
 
 		//
 
-		const renderObject = this._objects.get( object, material, scene, camera, lightsNode );
-		renderObject.context = this._currentRenderContext;
+		object.modelViewMatrix.multiplyMatrices( camera.matrixWorldInverse, object.matrixWorld );
+		object.normalMatrix.getNormalMatrix( object.modelViewMatrix );
 
 		//
 
-		this._nodes.updateBefore( renderObject );
+		material.onBeforeRender( this, scene, camera, geometry, material, group );
 
 		//
 
-		object.modelViewMatrix.multiplyMatrices( camera.matrixWorldInverse, object.matrixWorld );
-		object.normalMatrix.getNormalMatrix( object.modelViewMatrix );
+		if ( material.transparent === true && material.side === DoubleSide && material.forceSinglePass === false ) {
+
+			material.side = BackSide;
+			this._renderObjectDirect( object, scene, camera, geometry, material, group, lightsNode, 'backSide' );
+
+			material.side = FrontSide;
+			this._renderObjectDirect( object, scene, camera, geometry, material, group, lightsNode );
+
+			material.side = DoubleSide;
+
+		} else {
+
+			this._renderObjectDirect( object, scene, camera, geometry, material, group, lightsNode );
```

Comment:
Maybe we should here also use `frontSide` or `backSide` namespace depending on material side? Not sure if I understood the namespaces approach correctly...

Reply:
Namespace can cause confusion, I renamed the term to `pass id`, seems more appropriate. It is compared with a parameter in chain map to get a unique `RenderObject`: 

```js
this._objects.get( object, material, scene, camera, lightsNode, passId )

...

chainMaps[ passId ].get( object, material, scene, camera, lightsNode );
```